--- a/httplib.py	2015-06-22 18:49:29.000000000 +0000
+++ b/httplib.py	2015-06-28 23:17:10.303029633 +0000
@@ -830,8 +830,13 @@
         # it will avoid performance problems caused by the interaction
         # between delayed ack and the Nagle algorithm.
         if isinstance(message_body, str):
-            msg += message_body
-            message_body = None
+            try:
+                msg += message_body
+                message_body = None
+            except UnicodeDecodeError:
+                # This could be binary data - we can treat it like
+                # something that isn't a str instance
+                pass
         self.send(msg)
         if message_body is not None:
             #message_body was not a string (i.e. it is a file) and
